### PR TITLE
Chore:  typography base styles

### PR DESCRIPTION
**Pull request by howema**
### Description

**Base Styles: Typography**

Settings:
Added a file to styles/settings that contains a map which covers all font-sizes used in figma.
Note- the detail-page-header is an outlier that is only used once, thus given its own differing variable and keeping the map more streamlined.
-Ranges from 3xs - 3xl + the detail-page-header
-Process- Began by gathering all font-sizes used in figma, while combining duplicates, then ordering from min-max. Finished by converting all from px to rem. Referenced seesparkbox and sparkle pages.

Added a file to styles/settings that contains a map which covers all font-families used in figma.
-Only contains 2 fonts

Tools:
Added a file to styles/tools that contains a function that makes using font-sizes easier
Added a file to styles/tools that contains a function that makes using font-families easier
Added additional file, at the advice of Jon, in the styles/tools folder that contains the stringify function needed for both functions inside the same file. This will later be added to a base.scss file and imported where necessary.

_Most recent card updates_
Added fonts folder to root directory to contain font files
-currently cannot find any free downloads of sole sans regular or extended

Added _fonts.scss to newly created styles/generic folder
-for any necessary @font-face declarations, which will be imported by the SCSS entrypoint
-have not created any of these because I don't have the fonts

### Spec

Designs: [Figma Project Pages](https://www.figma.com/file/kswpPSHH2h5njVyfGDbIHk/FSA-Capstone-Project-2022?node-id=0%3A1)

See Story: [FSA22V1-164](https://sparkbox.atlassian.net/browse/FSA22V1-164)

### Validation
* [ ] This PR has new SCSS functions, mixins, or variables, so [Sass Tests](https://seesparkbox.com/foundry/how_and_why_we_unit_test_our_sass) were added or updated, and they pass.
* [ ] Changes are browser tested. This includes functionality, accessibility, responsiveness, and design.

#### To Validate

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, Snyk, etc).
3. Pull down all related branches.
4. Confirm all tests pass.
5. Confirm all card content is present
6. Confirm all card content is correct

### CODE REVIEW

File: styles/generic/_fonts.scss [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// any necessary @font-face declarations,
+// and those are imported by the SCSS entrypoint
```
Comment (dustin-jw) [inline styles/generic/_fonts.scss:2]: You can delete this file. With the font licensing issues we discussed previously, we won't be self-hosting any fonts. See https://sparkbox.atlassian.net/browse/FSA22V1-205 for details.

File: styles/settings/font_family_map.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Font Family
+$font-families: ('sans': ("Sole Sans", Helvetica, sans-serif),
+    'extended': ("Sole Sans Extended", Helvetica, sans-serif));
```
Comment (dustin-jw) [inline styles/settings/font_family_map.scss:3]: There are a lot of SCSS linting errors. Most of them are auto-fixable , so you can run `npm run lint:scss -- --fix`. Then check the output for any remaining issues and you can fix those manually.

**Review by dustin-jw — CHANGES_REQUESTED** [review]

File: styles/generic/_fonts.scss (hunk 1 above)
Comment (howema) [reply styles/generic/_fonts.scss]: Awesome, got it done!

File: styles/settings/font_family_map.scss (hunk 2 above)
Comment (howema) [reply styles/settings/font_family_map.scss]: Made the font and linting changes! :)

File: styles/settings/font_size_map.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+//     '3xs': 13px,
+//     '2xs': 14px,
+//     '1xs': 16px,
+//     'sm': 17px,
+//     'md': 18px,
+//     'lg': 24px,
+//     '1xl': 32px,
+//     '2xl': 40px,
+//     '3xl': 48px,
+//     'detail-page-name': 56px);
```
Comment (dustin-jw) [inline styles/settings/font_size_map.scss:24]: I would rename this file to `_font-sizes.scss`. The leading underscore is a SCSS convention to indicate that the file is meant to be imported. `globals.scss` should be the only SCSS file that doesn't start with an underscore.

**Review by dustin-jw — CHANGES_REQUESTED** [review]

File: styles/tools/functions_test.scss [hunk 4]
```diff
@@ -1,9 +1,26 @@
...
 @include describe("color function") {
   @include it("returns hex value from colors scss map") {
     @include assert-equal(color(blue-100), #00abc5);
   }
 }
+
+@include describe("font-size function") {
+  @include it("returns rem value from font-size scss map") {
+    @include assert-equal(font-size(sm),
+    1.063rem);
```
Comment (dustin-jw) [inline styles/tools/functions_test.scss:16]: I would put all of this on one line, just for readability.

```suggestion
    @include assert-equal(font-size(sm), 1.063rem);
```

File: styles/tools/functions_test.scss [hunk 5]
```diff
@@ -1,9 +1,26 @@
...
+  }
+}
+
+@include describe("font-families function") {
+  @include it("returns list of fonts from font-families scss map") {
+    @include assert-equal(font-family("mono"),
+    (menlo, consolas, monaco)
+  );
```
Comment (dustin-jw) [inline styles/tools/functions_test.scss:24]: Same here.

```suggestion
    @include assert-equal(font-family("mono"), (menlo, consolas, monaco));
```

**Review by dustin-jw — APPROVED** [review]
I have a couple code style nitpicks, but I won't let that hold up the PR, since it's passing linting. @arnest00, feel free to add your review!

**Review by arnest00 — APPROVED** [review]
We're good to go!

**Pull request merged**